WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java

```diff
@@ -565,6 +564,10 @@ public void loginViaWpcomUsernameInstead() {
 
     @Override
     public void showMagicLinkSentScreen(String email, boolean allowPassword) {
+        if (!allowPassword) {
+            getSupportFragmentManager().popBackStackImmediate();
+        }
```

Comment:
We should move this logic to `LoginMagicLinkRequestFragment` and replace `allowPassword` with `mForceRequestAtStart`. This flag indicates if we're skipping the screen or not, so it's the best one to use in our case.

Then where we now have:

```java
if (mLoginListener != null) {
    mLoginListener.showMagicLinkSentScreen(mEmail, mAllowPassword);
}
```

 Should then be something like:

```java
if (mLoginListener != null) {
    if (mForceRequestAtStart) {
        getFragmentManager().popBackStackImmediate();
    }
    mLoginListener.showMagicLinkSentScreen(mEmail, mAllowPassword);
}
```

I'd probably also add a comment there, as it's not very obvious what's going on. Something like: 
`// We don't want the user to go back to this screen if they press back on the next one.`